pink/tasks/linear_holonomic_task.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright 2022 Stéphane Caron
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Linear Holonomic task implementation."""
+
+from typing import Optional, Tuple
+
+import numpy as np
+import pinocchio as pin
+
+from ..configuration import Configuration
+from ..utils import get_joint_idx
+from .exceptions import TargetNotSet
+from .task import Task
+
+
+class LinearHolonomicTask(Task):
```

Comment:
A general linear holonomic task would be:

$$
e(q) = A q - b
$$

$$
\dot{e}(q) = A \dot{q} \Rightarrow J(q) = A
$$

In this PR what you have implemented is a particular case (where $A$ couples two joints), but we could let the user supply any matrix $A$ to implement the general linear-holonomic task.

Reply:
I was thinking of ``__init__(self, A, cost)``, but indeed that delegates the computation of $A$ to the user and we lose some useful code from your use case.

How about making a general `LinearHolonomicTask`, and a child

```python
class JointCouplingTask(LinearHolonomicTask):
```

That builds `self.A` internally and gives users a nicer interface?

Currently the coupling is between two joints but we could imagine a function like ``task.add_coupling(["shoulder_pitch", "elbow", "wrist"], [1.0, -0.5, 1.0])`` that the task would internally convert into adding the proper line to ``self.A``.

Reply:
Awesome. That would work better. I'll work on that based on your suggestion.